images/podman/podman-compose-up/playbook.yml

```diff
@@ -0,0 +1,120 @@
+- name: Fetch compose environment secrets from vault
+  hosts: localhost
+  vars:
+    skip_compose_up: "{{ skip_start | default(false) }}"
+  tasks:
+    - name: check for config drive
+      stat:
+        path: /dev/sr0
+      register: config_drive
+
+    # load metadata from config drive
+    - block:
+        - name: Mount config drive
+          ansible.posix.mount:
+            path: /mnt/config-drive
+            src: /dev/sr0
+            fstype: iso9660
+            opts: ro
+            boot: false
+            state: mounted
+
+        - name: Get instance metadata from drive
+          ansible.builtin.set_fact:
+            metadata: "{{ lookup('file', '/mnt/config-drive/openstack/latest/meta_data.json') | from_json }}"
+
+        - name: Unmount config drive
+          ansible.posix.mount:
+            path: /mnt/config-drive
+            src: /dev/sr0
+            opts: ro
+            state: absent
+
+      when: config_drive.stat.exists
+
+    # load metadata from metadata service
+    - name: Get instance metadata from url
+      ansible.builtin.set_fact:
+        metadata: "{{ lookup('url', 'http://169.254.169.254/openstack/latest/meta_data.json') | from_json }}"
+      when: not config_drive.stat.exists
+
+    - name: Fetch Vault token
+      ansible.builtin.uri:
+        url: "{{ metadata.meta.vault_addr }}/v1/auth/openstack/login"
```

Comment:
It would be better if the path (i.e. vault mount) could be specified as well, I think.